pygmt/clib/session.py

```diff
@@ -2018,6 +2028,10 @@ def virtualfile_to_raster(
                 self["GMT_IS_IMAGE"]: "image",
                 self["GMT_IS_CUBE"]: "cube",
             }[family]
+        if kind == "image":
+            with GMTTempFile(suffix=".tif") as tmpfile:
+                self.call_module("write", f"{vfname} {tmpfile.name} -Ti")
+                return xr.load_dataarray(tmpfile.name)
```

Comment:
Ah I see, so this is just a placeholder for now. Yes, this PR will need to wait for #3128, though I'm thinking of cherry-picking some of the clib-related changes here in #3115 to #3128, or opening a separate PR for that.

Reply:
The original idea of this PR is to make `grdcut` support both grids (via GMT_GRID) and images (via temporary files because GMT_IMAGE is not wrapped yet). I can continue this PR without waiting for #3128. Maybe it's easier.

Reply:
That could work too. Finish up this PR #3115 with the placeholder code, and then we'll refactor things in #3128 to remove the tempfile path. Either way, we should get both of these more-or-less ready at the same time, so that we have both of them for PyGMT v0.13.0.

Reply:
I've decided to finish this PR without waiting for PR #3128. Two tests are also added which can help test changes in PR #3128, too.